### PR TITLE
[backport] document missing environment variables

### PR DESCRIPTION
**Pull request by niboshi**
Backport of #3662

### CODE REVIEW

**Comment by niboshi** [comment]
I removed the mention to `CHAINER_USE_IDEEP`.

**Comment by kmaehashi** [comment]
In v3 branch, it should be backported to `docs/source/reference/environment.rst`.

**Comment by niboshi** [comment]
Fixed.

File: docs/source/reference/core/configuration.rst [hunk 1]
```diff
@@ -68,7 +68,15 @@ Note that the default values are set in the global config.
        - If it is ``'auto'``, Chainer will use cuDNN only if it is known that the usage does not degrade the performance.
        - If it is ``'never'``, Chainer will never use cuDNN anywhere.
 
-   The default value is ``'auto'``.
+   The default value is given by ``CHAINER_USE_CUDNN`` environment variable (set to ``'always'``, ``'auto'`` or ``'never'``) if available, otherwise uses ``'auto'``.
+``chainer.config.cudnn_use_tensor_core``
```
Comment (kmaehashi) [inline docs/source/reference/core/configuration.rst]: TensorCore part should not be backported.

Comment (niboshi) [reply docs/source/reference/core/configuration.rst]: Thanks. Fixed.

**Comment by kmaehashi** [comment]
LGTM!

**Pull request merged**